packages/contracts-core/contracts/test/utils/NomadTest.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+
+import "forge-std/Test.sol";
+import "forge-std/console2.sol";
+import "../../UpdaterManager.sol";
+import "./BadXapps.sol";
+
+
+contract NomadTest is Test {
+    uint256 updaterPK = 1;
+    uint256 fakeUpdaterPK = 2;
+    address updater = vm.addr(updaterPK);
+    address fakeUpdater = vm.addr(fakeUpdaterPK);
+    address signer = vm.addr(3);
+    address fakeSigner = vm.addr(4);
+
+    uint32 homeDomain = 1500;
+    uint32 remoteDomain = 1000;
+
+    function setUp() public virtual {
+        vm.label(updater, "updater");
+        vm.label(fakeUpdater, "fake updater");
+        vm.label(signer, "signer");
+        vm.label(fakeSigner, "fake signer");
+    }
+
+    function getMessage(bytes32 oldRoot, bytes32 newRoot, uint32 domain)
```

Comment:
naming not ideal. maybe `prepUpdate` or something?

Reply:
To which function are you referring? Not sure I see where prepUpdate is relevant

Reply:
`getMessage` appears to be doing the update signing process's preparation. i.e serializing the update in a format suitable for signing w/ domain binding. not "getting the message" so much as "prepping the update for signing"